src/yardstick/cli/config.py

```diff
@@ -63,6 +66,53 @@ def __post_init__(self):
             else:
                 images += [image]
         self.images = images
+        invalid = [
+            image for image in images if not ScanMatrix.is_valid_oci_reference(image)
+        ]
+        if invalid:
+            raise ValueError(
+                f"all images must be complete OCI references, but {' '.join(invalid)} are not"
+            )
+
+    @staticmethod
+    def is_valid_oci_reference(image: str) -> bool:
+        host, _, repository, _, digest = ScanMatrix.parse_oci_reference(image)
+        return (
+            all([host, repository, digest])
+            and bool(ScanMatrix.DIGEST_REGEX.match(digest or ""))
+            and ("." in host or "localhost" in host)
+        )
+
+    @staticmethod
+    def parse_oci_reference(image: str) -> tuple[str, str, str, str, str]:
```

Comment:
this seems like something for which [a library already exists](https://github.com/realityone/docker-image-py); would it be worthwhile to use one instead of building our own parser here?

Reply:
Thanks for the suggestion! I tried out this library, but I don't think we should include it for a few reasons:

1. It doesn't have any type hinting, so it's hard to integrate
2. It has a class hierarchy of different types of references and types of exceptions, which makes it a little unwieldy, especially given the lack of type hinting
3. It doesn't split things up the way I'd hoped (doesn't split off a domain and a path)

``` py
>>> from docker_image import reference
>>> reference.Reference.parse("registry.access.redhat.com/ubi8@sha256:68fecea0d255ee253acbf0c860eaebb7017ef5ef007c25bee9eeffd29ce85b29")
{'name': 'registry.access.redhat.com/ubi8', 'tag': None, 'digest': 'sha256:68fecea0d255ee253acbf0c860eaebb7017ef5ef007c25bee9eeffd29ce85b29'}
```

Notice that `name` is both the hostname and the path, but validating that there is both a hostname and path is the point of this change, so I would just have to further parse whatever this library is doing.

It _sort of_ does conveys the information we want via the class of the return value of parse, `CanonicalReference`, `Reference`, `TaggedReference`, but I don't want to rely on checking return types. More importantly, things get demoted from `CanonicalReference` to `Reference` if we include a tag, but we want to include a tag so that humans can figure out what we meant by the thing.